.github/workflows/sonarcloud.yml

```
@@ -1,188 +1,65 @@
   build-and-analyze:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout repository
         uses: actions/checkout@v4
         with:
-          fetch-depth: 0  # Needed for SonarCloud branch analysis
-      
+          fetch-depth: 0
+
       - name: Set up JDK 21
         uses: actions/setup-java@v3
```

actionlint (1.7.4)

the runner of "actions/setup-java@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)